common/src/sql/dbinit.sql

```diff
@@ -1419,6 +1419,98 @@ CREATE INDEX ON omicron.public.update_available_artifact (
     targets_role_version
 );
 
+/*
+ * System updates
+ */
+CREATE TABLE omicron.public.system_update (
+    /* Unique semver version */
+    version STRING(64) PRIMARY KEY,
```

Comment:
If we're representing semver versions as STRING types, we won't be able to compare them within DB operations, right? 

For example, let's suppose we have two `system_update` objects, and we want to order them. Maybe we want to paginate by version, maybe we just want to query if `A < B`.

We can't perform a lexicographic comparison here, since that would make a version like `1.11.0` appear smaller than `1.9.0`. Do we need to identify semver versions by subcomponents?

Reply:
Definitely a problem. I talked to Justin about it and mostly decided to punt. I don't really know how to do it. Storing the components as separate columns sounds like a huge pain in the ass, but maybe Diesel can help. Another approach would be to generate a stable, lexicographically (or numerically? probably not possible with the string meta bits) sortable representation to store alongside the original. I think it would require us to commit to limits on how big each number can be. so for example we could store `1.22.3` as `"00000001.00000022.00000003"` or something.

Reply:
Clever; that padding approach could probably get us pretty far.

Diesel does have some support for embedded structures - that's what I'm doing all over the place with the `Identity` objects that are derived for assets / resources - but it's definitely a little more cumbersome than just a single value.